Extension/src/Utility/Eventing/eventParser.ts

```diff
@@ -0,0 +1,270 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import { hasErrors } from '../Sandbox/interfaces';
+import { Sandbox } from '../Sandbox/sandbox';
+import { first } from '../System/array';
+import { smash } from '../Text/identifiers';
+import { Kind, Scanner, Token } from '../Text/scanner';
+import { verbose } from '../Text/streams';
+import { ArbitraryObject, Filter } from './interfaces';
+
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+// ensure that array gets the first function, we need it for the discriminator
+// eslint-disable-next-line no-unused-expressions
+first;
+
+function isKeyword(token: Token) {
+    return token.kind > Kind.KeywordsStart && token.kind < Kind.KeywordsEnd;
+}
+
+export function parse(triggerExpression: string, sourceToBindTo: ArbitraryObject | undefined): [boolean, boolean, Map<string, Filter | true>, ArbitraryObject?] {
+    //
+    // [modifiers] event[filter]:discriminator[filter]/discriminator[filter]/discriminator[filter]...
+    //
+
+    const scanner = new Scanner(triggerExpression).start();
+    let token: Token;
+
+    let once: boolean = false;
+    let source: any;
+    let isSync = false;
+    // const name= '';

```

Comment:
Did you want to remove this commented out line or is it a reminder for some future use case?